src/terminal/char_grid/cursor.rs

```diff
@@ -24,10 +24,20 @@ pub struct Cursor {
     pub coords: Coords,
     pub style: Styles,
     pub text_style: Styles,
+    config: Config,
```

Comment:
Scratch that last comment, the clever thing I wanted to do is actually unsound.

We can't store the config reference on the child structs, which means we have to pass it in as an argument to every function that needs it, like you've done with the renderers. This is a lot of boilerplate but its how it works. But wait - there's more!

So far, in order to be able to iterate quickly, I've had `Terminal` deref to `CharGrid`, so I didn't have to write delegation methods. Not sure if you know this, but when you call a method like `terminal.move_cursor`, Rust will insert as many dereference operators before `terminal` as it needs to do get to a type that has a `move_cursor` method.

However, now the signature of a lot of `CharGrid`'s methods will change to add the config argument. That means we need explicit delegations from `Terminal` to `CharGrid` which pass in `Terminal`'s config struct. Sorry, lot's of boilerplate.